subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -409,9 +431,17 @@ static void scene_delete(struct bt_mesh_scene_srv *srv, uint16_t *scene)
 		(void)bt_mesh_model_data_store(srv->mod, false, path, NULL, 0);
 	}
 
-	if (srv->curr == *scene) {
+	int64_t now = k_uptime_get();
+
+	if (target_scene(srv, now) == *scene ||
+	    (current_scene(srv, now) == *scene &&
+	     target_scene(srv, now) == BT_MESH_SCENE_NONE)) {
 		scene_set(srv, BT_MESH_SCENE_NONE);
 		srv->transition_end = 0U;
+		srv->prev = BT_MESH_SCENE_NONE;
+	} else if (current_scene(srv, now) == *scene &&
+		   target_scene(srv, now) != *scene) {
+		srv->prev = BT_MESH_SCENE_NONE;
```

Comment:
Create temporary target and current variables, so we don't have to do the function evaluation over and over. It'll also make the ifs look a bit cleaner.